docs/content/preview/architecture/transactions/transaction-priorities.md

```diff
@@ -0,0 +1,180 @@
+---
+title: Transaction priorities
+headerTitle: Transaction priorities
+linkTitle: Transaction priorities
+description: Details about Transaction priorities in YSQL
+menu:
+  preview:
+    identifier: architecture-transaction-priorities
+    parent: architecture-acid-transactions
+    weight: 1153
+type: docs
+---
+
+<ul class="nav nav-tabs-alt nav-tabs-yb">
+
+  <li >
+    <a href="../transaction-priorities/" class="nav-link active">
+      <i class="icon-postgres" aria-hidden="true"></i>
+      YSQL
+    </a>
+  </li>
+
+</ul>
+
+When using the [Fail-on-Conflict](../concurrency-control/#fail-on-conflict) concurrency control policy, transactions are assigned priorities that help decide which transactions should be aborted on conflicts.
+
+For distributed transactions, the priority space is `unit64_t`. The 64 bit range is split into 2 priority buckets -
+
+1. Normal priority bucket: `[yb::kRegularTxnLowerBound, yb::kRegularTxnUpperBound]` i.e., 0 to  `uint32_t_max`-1
+2. High priority bucket: `[yb::kHighPriTxnLowerBound, yb::kHighPriTxnUpperBound]` i.e., `uint32_t_max` to `uint64_t_max`
+
+All transactions are usually randomly assigned a priority in the first bucket (normal). However, in case a transaction has the first query which takes a `FOR UPDATE/ FOR SHARE/ FOR NO KEY UPDATE` explicit row lock using SELECT, it is randomly assigned a priority from the high priority bucket.
+
+Apart from the above rule, there are two other user configurable session variables that can help control the priority assigned to transaction is a specific session. These are `yb_transaction_priority_lower_bound` and `yb_transaction_priority_upper_bound`. These help set lower and upper bounds on the randomly assigned priority a transaction should receive from the respective bucket that applies to it. For ease of use, the bounds are expressed as a float such that the numerical ranges of a bucket are proportionally map to floats from 0-1. Also note that the same floating point bounds apply to both buckets.
+
+For example, if `yb_transaction_priority_lower_bound=0.5` and `yb_transaction_priority_upper_bound=0.75`:
+
+1. a transaction that is assigned a priority from the normal bucket will get one between the 0.5-0.75 marks such that the `[yb::kRegularTxnLowerBound, yb::kRegularTxnUpperBound]` range proportionally maps to 0-1.
+
+2. a transaction that is assigned a priority from the high priority bucket will get between the 0.5-0.75 marks such that the `[yb::kHighPriTxnLowerBound, yb::kHighPriTxnUpperBound]` range proportionally maps to 0-1.
```

Comment:
these examples should probably be more explicit, e.g. specify what the actual ranges are which we would end up selecting from in each case, in the uint64 space, if the goal is to add clarity